### PR TITLE
[flang] Lower function/subroutine attribute to func op

### PR DESCRIPTION
**Pull request by clementval**
Keep track of the Fortran procedure attributes on the func operation.

### CODE REVIEW

File: flang/lib/Lower/Bridge.cpp [hunk 1]
```diff
@@ -5025,6 +5025,20 @@ class FirConverter : public Fortran::lower::AbstractConverter {
...
+        func.getOperation()->setAttr(fir::getFuncPureAttrName(),
+                                     builder->getUnitAttr());
+      if (IsElementalProcedure(funcSym))
+        func.getOperation()->setAttr(fir::getFuncElementAttrName(),
+                                     builder->getUnitAttr());
+      if (funcSym.attrs().test(Fortran::semantics::Attr::RECURSIVE))
+        func.getOperation()->setAttr(fir::getFuncRecursiveAttrName(),
+                                     builder->getUnitAttr());
+    }
+
```
Comment (jeanPerier) [inline flang/lib/Lower/Bridge.cpp:5041]: Can you move the logic to `addSymbolAttribute` https://github.com/llvm/llvm-project/blob/c4a1440c149d3ea03f14fd6858b6be3a2faf9af6/flang/lib/Lower/CallInterface.cpp#L582C13-L582C31 ?

**Review by jeanPerier — COMMENTED** [review]
Thanks, aside from the location, logic looks good to me.

**Comment by clementval** [comment]
> Thanks, aside from the location, logic looks good to me.

Yeah sure it makes more sense

**Review by vzakhari — APPROVED** [review]
LGTM!

**Pull request merged**